.circleci/config.yml

```diff
@@ -0,0 +1,153 @@
+defaults:
+  image: &image
+    docker:
+      - image: circleci/node:11.6.0
+  yarn_cache: &yarn_cache
+    key: yarn_cache-v{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ checksum "yarn.lock" }}
+  yarn_install: &yarn_install
+    name: Install
+    command: yarn
+  setup_git: &setup_git
+    name: Setup Git user
+    command: |
+      git config --global user.email "ops@groupbyinc.com"
+      git config --global user.name "GroupBy Ops"
+      git config --global push.default simple
+
+      cp .circleci/prepare-commit-msg .git/hooks/prepare-commit-msg
+      chmod +x .git/hooks/prepare-commit-msg
+  persist_repo: &persist_repo
+    root: ~/
+    paths:
+      - project
+      - .ssh
+
+version: 2
+
+jobs:
+  build:
+    <<: *image
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/
+      - restore_cache: *yarn_cache
+      - run:
+          name: Dependencies
+          command: sudo apt install -y rsync
+      - run: *yarn_install
+      - save_cache:
+          <<: *yarn_cache
+          paths:
+            - ~/.cache/yarn
+      - run:
+          name: Build
+          command: yarn build
+      - persist_to_workspace: *persist_repo
+
+  release:
+    <<: *image
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run: *setup_git
+      - run:
+          name: Install dependencies
+          command: sudo apt install -y ed
+      - run:
+          name: Release
+          command: |
+            if ./scripts/release.sh; then
+              exit_code=0
+            elif ec=$?; (( ec == 4 )); then
+              exit_code=0
+            else
+              exit_code=$ec
+            fi
+
+            exit $exit_code
+      - persist_to_workspace: *persist_repo
+
+  publish:
+    <<: *image
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: Install utilities
+          command: sudo apt install jq
+      - run:
+          name: Authenticate with registry
+          command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/.npmrc
+      - run:
+          name: Publish
+          command: |
+            if ! npm view ".@$(jq -r .version package.json)" | grep -q .; then
+              ./scripts/publish.sh
+            else
+              echo "Nothing to publish."
+            fi
+
+  docs:
+    <<: *image
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/
+      - restore_cache: *yarn_cache
+      - run: *yarn_install
+      - run: *setup_git
+      - run:
+          name: Docs
+          command: |
+            rm -rf docs
+            yarn docs
+      - run:
+          name: Replace files in gh-pages branch
+          command: |
+            mv docs ../
+            git checkout gh-pages
+            git rm "*"
+            shopt -s dotglob
+            rm -rf esnext node_modules
```

Comment:
Consider doing a `git clean` (with the appropriate flags instead. Or since we want everything but `.git` you can do the following:

```bash
shopt -s extglob
rm -r -- !(.git)
```

This will remove everything but the `.git` directory.

If you decide to keep the two commands, put the two `rm` commands together.

Reply:
If we're already avoiding the `.git` folder, why not also avoid the `docs` folder and save from moving it up and down?

Reply:
Will switch to `git clean -df`.

@DanielMoniz-GBI The `docs` folder needs to be flattened by one level as well so moving it out and then just the contents back in accomplishes that.